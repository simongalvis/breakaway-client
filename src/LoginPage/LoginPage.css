.LoginPage{
    display: flex;
    flex-direction:column;
    background-color:rgb(41, 41, 41);
    background-image: url(../images/running-in-snow.jpg);
    min-height: 520px;
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
}
.loginFormContainer{
    max-width: 550px;
    align-self: center;
    margin: 10% 0;
    
}
form{
    align-self: center;
    padding: 16px 0;
    margin-right:0;
    text-align:0;
    display: block;
    max-width:500px;
    background-color:rgb(41, 41, 41);
    
    
}
input{
    width:80%;
    margin:3%;
    border-radius: 7px;
    border:solid 1px gray;
    height:21px;
    background-color:black;
    max-width:700px;
}
#loginheading{
    color:white;
    background-color: black;
    margin:0;
    padding: 24px 24px;
}
#loginbutton{
    width: 80%;
    margin-top:4%;
    border-radius:33px;;
    background-color:rgb(13, 146, 255);
    color:white;
    border: 1px solid rgb(41, 41, 41);
}

@media screen and (min-width: 500px){
    .loginFormContainer{
        width:454px;
    }
    }