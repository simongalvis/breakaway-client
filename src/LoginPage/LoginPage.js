import React from 'react';
import '../LoginPage/LoginPage.css';
import LoginNavBar from '../LoginNavBar/LoginNavBar.js';
import ApiContext from '../ApiContext';




class LoginPage extends React.Component{
static contextType = ApiContext;

handleSubmit = e =>{
    e.preventDefault();
   

const  { loginUsername, loginPassword } = e.target;
 
this.context.loginUser(loginUsername.value, loginPassword.value )
   
  
   
setTimeout( () => {if (Object.keys(this.context.loggedUser).length > 0){
           this.props.history.push('/activities')}}, 500)
   
   




}
    render(){
        return(
            <div className='LoginPage'>
                <LoginNavBar />
                <div className='loginFormContainer'>
                <h2 id='loginheading'>Log In</h2>
                <form onSubmit={this.handleSubmit}>
                    <input  type='text' id='loginUsername' name='loginUsername' placeholder='Username' required/>
                    <input  type='password' id='loginPassword' name='loginPassword' placeholder='Password' required/>
                        <button type="submit" id='loginbutton'>Log In</button>
                    
                </form>
                </div>
                 
                
                
            </div>
            
        )
    }
}

export default LoginPage;




