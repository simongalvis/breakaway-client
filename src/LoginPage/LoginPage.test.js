import React from 'react'
import ReactDOM from 'react-dom'
import LoginPage from './LoginPage'
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom'

describe(`LoginPage component`, () =>{
  it(`renders without crashing`, () =>{
    const div = document.createElement('div');
  
    ReactDOM.render(<BrowserRouter><LoginPage/></BrowserRouter>, div);
  
    ReactDOM.unmountComponentAtNode(div);
  })
  it(`Renders UI as expected`, () =>{
    const tree = renderer
      .create(<BrowserRouter><LoginPage/></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})
