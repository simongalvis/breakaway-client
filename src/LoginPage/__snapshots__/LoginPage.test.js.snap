// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component Renders UI as expected 1`] = `
<div
  className="LoginPage"
>
  <header>
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Blue breakaway logo"
        id="main-logo"
        src="breakaway-logo.png"
      />
    </a>
    <a
      href="/signup-page"
      onClick={[Function]}
    >
      <button
        className="nav-button"
        id="signup-button"
        value="signup"
      >
        Sign Up
      </button>
    </a>
  </header>
  <div
    className="loginFormContainer"
  >
    <h2
      id="loginheading"
    >
      Log In
    </h2>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="loginUserName"
      >
        <b>
          Username:
        </b>
      </label>
      <input
        autoCapitalize="none"
        autoCorrect="off"
        id="loginUsername"
        name="loginUsername"
        placeholder="Username"
        required={true}
        type="text"
      />
      <label
        htmlFor="loginPassword"
      >
        <b>
          Password:
        </b>
      </label>
      <input
        autoCapitalize="none"
        autoCorrect="off"
        id="loginPassword"
        name="loginPassword"
        placeholder="Password"
        required={true}
        type="password"
      />
      <button
        id="loginbutton"
        type="submit"
      >
        Log In
      </button>
    </form>
  </div>
</div>
`;
